posts/2023-12-31-foss-engines-overview.md

```diff
@@ -0,0 +1,112 @@
+
+### Bevy
+* Written in Rust, Supported Languages: Rust
+* Platforms:
+    * Development (no Editor): Windows, Mac, Linux
+    * Target: Windows, Linux, Mac, iOS, WASM
+
+As the most popular Rust-based game engine, Bevy offers a rich code-only development environment (an editor is coming) capable of running on all major operating systems (Windows, Mac, and Linux). At the heart of Bevy’s vision for game development lies the Entity Component System (ECS) paradigm. While there are other resources available which can explain the benefits of ECS better, in a nutshell, ECS breaks down the code into three core pillars: entities, components, and systems. Entities are composed of components which can then interact with each other using systems. For example, the player character could be an entity with a health component which tracks the player character’s health bar. An enemy character could also use that component for the same purpose. ECS encourages modularity and reusability by enabling developers to create components applicable to distinct entities. While other game engines can approximate a similar system, Bevy makes this part of its core design ethos.
+
+As established with Bevy’s use of ECS, the engine’s developers care deeply about modularity. The engine’s plugin system accentuates their commitment to that principle in every part of Bevy. Developers can organize the game’s systems into discrete plugins. An example of this is organizing all the UI systems code into a UI plugin. From there, developers can slot the plugin into the game’s initialization step. This helps organize the code, while also encouraging modularity by allowing the developer to add or remove the plugin based on their needs. This paradigm even applies to the engine’s core features, as they are all organized into plugins. It becomes trivial to activate or deactivate any part of the engine—including rendering, audio playback, and event loops—as the developer sees fit.
+
+Asset libraries provide a wealth of resources that empower developers to learn the tools quickly and get their game to a playable state. The community assembled a library of assets available on the official website for the community to use and share. The library includes tutorials, plugins, and templates that address subjects like physics, networking, and input management. There are even entire games and applications available in the asset library to either build on or use as a reference while learning the engine. Bevy’s structure encourages developers to use any of the resources from this library freely as part of the building blocks that will make up their game. In conjunction with Rust’s package manager, there is a strong emphasis on modularity at every level of the engine. 
+
+
+### Godot
+* Written in C++, Supported Languages: GDScript, C#
+* Platforms:
+    * Development: Windows, Linux, Mac
```

Comment:
You can add `Android (experimental)`

https://godotengine.org/download/android/

Reply:
And Web - https://editor.godotengine.org/releases/4.2.1.stable/godot.editor.html (with some limitation).

Reply:
Added